TASK: Add missing fusion migrations and runtime helper for 9.0

mhsdesign opened the pull request:
<!-- 
    Thanks for your contribution, we appreciate it!

    The first section should explain briefly what is changed. 
    Some examples are always nice to showcase the use. 
    The content will be used in the change-logs and addresses 
    developers working with Neos.

    If there are issues regarding the topic of your PR link 
    them here as `related:` or `resolved:`
-->

Followup to https://github.com/neos/neos-development-collection/pull/5638

I did a full re-review on all the Fusion migrations that were copied from `neos/rector`

The Fusion migrations were refined to ensure we handle even more cases (and in 9.0 write comments for all the things)

Doing that i noticed we were missing the 8.4 api for `node.workspaceName` and `node.classification.tethered` and `q(node).property("hiddenInMenu")` and ~enableAfterDateTime, disableAfterDateTime~ so i added the missing runtime layers for those too.

**Upgrade instructions**

<!-- 
    Add upgrade instructions for breaking changes. 
    Explain who is affected, what has to be adjusted  
-->

**Review instructions**

<!-- 
    If your change is not explained fully by the first section you can 
    add more details here to help the reviewers understand the change.
    We have to understand what you did, why you did it and how we can 
    verify it works correctly and does no harm.
-->

**Checklist**

- [ ] Code follows the PSR-2 coding style
- [ ] Tests have been created, run and adjusted as needed
- [ ] The PR is created against the [lowest maintained branch](https://www.neos.io/features/release-roadmap.html)
- [ ] Reviewer - PR Title is brief but complete and starts with `FEATURE|TASK|BUGFIX`
- [ ] Reviewer - The first section explains the change briefly for change-logs
- [ ] Reviewer - Breaking Changes are marked with `!!!` and have upgrade-instructions

mhsdesign committed:
PATCH: Remove un-necessary bloat from tests

mhsdesign committed:
BUGFIX: Ensure fusion migrations handles `node.workspace.name` and warns when `node.context` is passed around

mhsdesign committed:
BUGFIX: Only replace `.context.currentSite` when certain it's a node

mhsdesign committed:
TASK: Migrate node.tethered and node.nodeName from TraversableNodeInterface

mhsdesign committed:
TASK: Migrate node.propertyNames and add warning comments for remaining node getters

mhsdesign committed:
TASK: Add comment when sort operation is used with underscores

see https://github.com/neos/neos-development-collection/pull/5015

mhsdesign committed:
FEATURE: Ensure node.classification.tethered works in Fusion

mhsdesign committed:
FEATURE: Forward-compatibility of node properties enableAfterDateTime, disableAfterDateTime, hiddenInMenu

we migrate those from _hiddenBeforeDateTime
                      _hiddenAfterDateTime
                      _hiddenInIndex

mhsdesign committed:
FEATURE: Forward-compatibility of node.workspaceName

which is also available in Neos 9.0 while the workspace is not

mhsdesign committed:
TASK: Move forward compatibility layer into property operation instead of obscuring getProperty further

mhsdesign committed:
FEATURE: Improve core migration for legacy fusion objects by showing warnings in output and add Neos.Array.toHtmlAttributesString() forward compatibility

mhsdesign committed:
TASK: Emit warnings in core migration also for EEL replacements and use TODO Neos.Neos-20251005080230 instead of 9.0 migration as header

mhsdesign committed:
TASK: Ensure migration does not add comments over and over again when rerun

dlubitz commented:
I'm not sure if it makes sence to rewrite `enableAfterDateTime` and `disableAfterDateTime` as this is extracted into a dedicated package.

mhsdesign commented:
Well you thought of this as useful when you build that :D https://github.com/neos/rector/pull/42

But i understand that this migration to use real properties does not fix anything alone when using neos 9 ... you still have to use the package.

If we dont want that i can remove that part from the 8.4 migration and keep it in 9.0 ... or i can also remove that in 8.4 and 9.0 ... or we could add a comment ... or .. or . or

mhsdesign committed:
PATCH: Dont warn when `props.label` is used

mhsdesign committed:
PATCH: Prevent migration from failing for not parse-able fusion files and emit warning instead

mhsdesign committed:
BUGFIX: Prevent migrating `props.node` to something else like `props.renderingMode`

prevents migrating `${props.node.context.inBackend}` to `${props.renderingMode.isEdit}`

instead a warning is emitted

mhsdesign committed:
PATCH: Go safe route and dont migrate `this.node` as "node" is not surely a NEOS node

also that migration was only done for `this.node.depth` -> if we want to migrate that, then for all ;)

mhsdesign committed:
PATCH: Handle props.node.parent correctly

mhsdesign committed:
TASK: Dont migrate getHiddenBeforeDateTime and getHiddenAfterDateTime for neos 8.4

see https://github.com/neos/neos-development-collection/pull/5641#issuecomment-3382911880

mhsdesign committed:
TASK: Show migration warnings in `neos9preparation:preadjustfusion` if any

> File ./Tests/Functional/Fusion/Fixtures/Base.fusion: "Neos.Neos:PrimaryContent" has been removed without a complete replacement. We replaced all usages with "Neos.Neos:ContentCollection" but not the prototype definition. Please check the replacements and if you have overridden the "Neos.Neos:PrimaryContent" prototype and rewrite it for your needs.

or parser errors:

mhsdesign commented:
I made the adjustment to not migrate the timed hidden things. Also i fixed a bug where the migration would migrate too eagerly when finding `node` as in `props.node.label` which must not become `props.Neos.Node.label(node)`. INstead a warning is emitted and nothing is done.
